xdem/spstats.py

```diff
@@ -0,0 +1,507 @@
+
+    return (psill1 + psill2)/std_err**2
+
+def neff_circ(area: float,list_vgm: list) -> float:
+    """
+    Number of effective samples derived from numerical integration for any sum of variogram models a circular area
+    (generalization of Rolstad et al. (2009): http://dx.doi.org/10.3189/002214309789470950)
+    The number of effective samples N_eff serves to convert between standard deviation/partial sills and standard error
+    over the area: SE = SD / sqrt(N_eff) if SE is the standard error, SD the standard deviation.
+
+    :param area: area
+    :param list_vgm: variogram functions to sum
+
+    :returns: number of effective samples
+    """
+
+    psill_tot = 0
+    for vario in list_vgm:
+        psill_tot += vario[2]
+
+    def hcov_sum(h):
+        fn = 0
+        for vario in list_vgm:
+            crange, model, psill = vario
+            fn += h*(cov(h,crange,model=model,psill=psill))
+
+        return fn
+
+    h_equiv = np.sqrt(area / np.pi)
+
+    full_int = integrate_fun(hcov_sum,0,h_equiv)
+    std_err = np.sqrt(2*np.pi*full_int / area)
+
+    return psill_tot/std_err**2
+
+
+def neff_rect(area: float,width: float,crange1: float,psill1: float,model1: str ='Sph',crange2: float = None,
+              psill2: float = None,model2: str = None) -> float:
+    """
+    Number of effective samples derived from numerical integration for a sum of 2 variogram functions over a rectangular area
+
+    :param area: area
+    :param width: width of rectangular area
+    :param crange1: correlation range of first variogram
+    :param psill1: partial sill of first variogram
+    :param model1: model of first variogram
+    :param crange2: correlation range of second variogram
+    :param psill2: partial sill of second variogram
+    :param model2: model of second variogram
+
+    :returns: number of effective samples
+    """
+
+    def hcov_sum(h,crange1=crange1,psill1=psill1,model1=model1,crange2=crange2,psill2=psill2,model2=model2):
+
+        if crange2 is None or psill2 is None or model2 is None:
+            return h*(cov(h,crange1,model=model1,psill=psill1))
+        else:
+            return h*(cov(h,crange1,model=model1,psill=psill1)+cov(h,crange2,model=model2,psill=psill2))
+
+    width = min(width,area/width)
+
+    full_int = integrate_fun(hcov_sum,0,width/2)
+    bin_int = np.linspace(width/2,area/width,100)
+    for i in range(len(bin_int)-1):
+        low = bin_int[i]
+        upp = bin_int[i+1]
+        mid = bin_int[i] + (bin_int[i+1]- bin_int[i])/2
+        piec_int = integrate_fun(hcov_sum, low, upp)
+        full_int += piec_int * 2/np.pi*np.arctan(width/(2*mid))
+
+    std_err = np.sqrt(2*np.pi*full_int / area)
+
+    if crange2 is None or psill2 is None or model2 is None:
+        return psill1 / std_err ** 2
+    else:
+        return (psill1 + psill2) / std_err ** 2
+
+
+def integrate_fun(fun: Union[function,LowLevelCallable],low_b: float ,upp_b: float) -> float:
```

Comment:
Functions are `Callable`'s in python. Anything that you can put a `()` after is a callable.

```python
from typing import Callable

def apply_function(values: list[float], function: Callable) -> list[float]:
    return [function(value) for value in values]

Reply:
`list[subtype]` is a default syntax since python=3.9, but can be used in previous versions by including `from __future__ import annotations`. 

I've found that mypy (a static type checker) doesn't understand this in the latest release! Therefore, I install mypy from source (`pip install git+https://github.com/python/mypy.git`), which has this addition. I don't know from where PyCharm gets its type checks, but maybe it's a setting somewhere? 

Either way, I would consider that behaviour a bug, since your syntax is perfectly valid since maybe python=3.5 or something.

Reply:
Indeed everything runs perfectly with `from __future__ import annotations`.
I'll wait for PyCharm to correct the syntax correction bug. Thanks!